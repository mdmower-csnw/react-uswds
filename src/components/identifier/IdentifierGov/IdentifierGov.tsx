import React, { type JSX } from 'react'
import classnames from 'classnames'

export type IdentifierGovProps = {
  children?: React.ReactNode
  className?: string
} & JSX.IntrinsicElements['section']

export const IdentifierGov = ({
  children,
  className,
  ...sectionProps
}: IdentifierGovProps): JSX.Element => {
  const classes = classnames(
    'usa-identifier__section usa-identifier__section--usagov',
    className
  )

  return (
    <section data-testid="identifierGov" className={classes} {...sectionProps}>
      <div className="usa-identifier__container">{children}</div>
    </section>
  )
}

export default IdentifierGov
