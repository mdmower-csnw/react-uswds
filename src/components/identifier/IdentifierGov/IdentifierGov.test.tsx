import React from 'react'
import { render } from '@testing-library/react'

import { IdentifierGov } from './IdentifierGov'

describe('IdentifierGov component', () => {
  it('renders without errors', () => {
    const { queryByTestId } = render(<IdentifierGov />)
    expect(queryByTestId('identifierGov')).toBeInTheDocument()
  })

  it('renders section attributes passed in through props', () => {
    const { queryByTestId } = render(
      <IdentifierGov aria-label="custom aria-label value" />
    )

    expect(queryByTestId('identifierGov')).toHaveAttribute(
      'aria-label',
      'custom aria-label value'
    )
  })

  it('renders consistently in Spanish', () => {
    const { container } = render(
      <IdentifierGov>
        <div className="usa-identifier__usagov-description">
          ¿Necesita información y servicios del Gobierno?
        </div>
        &nbsp;
        <a href="https://www.usa.gov/espanol/" className="usa-link">
          Visite USAGov en Español
        </a>
      </IdentifierGov>
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
