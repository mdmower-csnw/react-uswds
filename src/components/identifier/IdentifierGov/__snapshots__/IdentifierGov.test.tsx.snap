// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`IdentifierGov component > renders consistently in Spanish 1`] = `
<section
  class="usa-identifier__section usa-identifier__section--usagov"
  data-testid="identifierGov"
>
  <div
    class="usa-identifier__container"
  >
    <div
      class="usa-identifier__usagov-description"
    >
      ¿Necesita información y servicios del Gobierno?
    </div>
     
    <a
      class="usa-link"
      href="https://www.usa.gov/espanol/"
    >
      Visite USAGov en Español
    </a>
  </div>
</section>
`;
